lib/omniauth/strategies/openid_connect.rb

```diff
@@ -312,13 +317,36 @@ def id_token_callback_phase
         call_app!
       end
 
+      def type_response?(type)
+        if options.response_type.is_a?(Array)
+          options.response_type.include?(type.to_s) || options.response_type.include?(type.to_sym)
+        else
+          options.response_type.to_s == type.to_s
+        end
+      end
+
+      def valid_param?(name)
+        params[name.to_s] && type_response?(name)
+      end
+
       def valid_response_type?
-        return true if params.key?(options.response_type)
+        Array(options.response_type).each do |response_type|
+          next if valid_param?(response_type)
 
-        error_attrs = RESPONSE_TYPE_EXCEPTIONS[options.response_type.to_s]
-        fail!(error_attrs[:key], error_attrs[:exception_class].new(params['error']))
+          error_attrs = RESPONSE_TYPE_EXCEPTIONS[response_type.to_s]
+          fail!(error_attrs[:key], error_attrs[:exception_class].new(params['error']))
 
-        false
+          return false
+        end
+
+        true
+      end
+
+      def verify_id_token(id_token)
```

Comment:
what's the point to pass `if_token` as argument here? can't you access it through `params`?